src/build_tools.md

```diff
@@ -0,0 +1,122 @@
+# Build Tools
```

Comment:
In the hindsight I'm not sure if "Development Tools" would not be a better name but I'm willing to crowd source here. What do you think @jaemk ?

Reply:
Now that I'm thinking about it, I'm wondering what other types of examples (other than `cc` stuff) will eventually be included here, if any. Does something of this nature deserve a more specifically named category? Maybe one of the following?
- `External Build Tools`
- `Custom Build Steps`
- `Building FFI Libraries`
- `Compiling and Integrating Non-Rust Code`
- `Build Script Support` (the conventional name [used by cargo](http://doc.crates.io/build-script.html))

If not, I think `Development Tools` might be better for covering a broader topic-space.

Reply:
Hmm in time we might want to have here other types of support tooling such as
- [`syn`](https://crates.io/crates/syn) 
- [`cmake`](https://crates.io/crates/cmake)
- [`pkg-config`](https://crates.io/crates/pkg-config)

so imho `Development Tools`, `Compile time` or `Build time dependencies` might be appropriate (I'm leaning towards the last one). But I'll let you chose, its easy enough to change anytime ;)

Reply:
What do you think of `Build Time Tooling` or `Build Time Tools`?